### DISCUSSION TITLE
Links and Text from Browser directly to ZIM.

### DISCUSSION THREAD

**Discussion by OPUltra** [zim-desktop-wiki/zim-desktop-wiki#2448] [Ideas]
Hello I only use ZIM for my private life and it is so super.

It would be very helpful to me to get a plugin to link from web pages and if necessary even highlighted text to ZIM. Is this possible? Honestly a big thank you to all who have developed ZIM. Super.

Bye

**Comment by brucewestfall** [comment]
Can you include some details on how it would be used?  Some examples would be very helpful.  For example zim can easily do links to specifid site, local folders and individual files.  Might be something like having a link to a script someone has already made.  Lots of smart people on here and perhaps your detailed idea could spark some amazing things!

I use ZIM many times a day and am always interested in how to expand it.

Looking forward to hearing more about your idea!

**Comment by introt** [comment]
I use the build-in web server for linking from eg. TiddlyWiki to my Zim notes, but you could also use eg. `zimwiki://` links with a custom handler to open the linked page source text file with Zim. It'd help if you could elaborate on your desired workflow.

**Reply by introt** [reply]
For the other direction there's eg. https://addons.mozilla.org/en-GB/firefox/addon/zim-clip/

**Comment by maclomhair** [comment]
I also think that such a feature would be really nice! It'd allow much tighter integration of e.g. spreadsheets or even jupyter notebooks with zim, as I'd be able to link from them back to the notebook. (Of course, the automatic link adjustment on page move would be very difficult to implement)

This is not implemented yet, AFAIK. zim can already handle `zim+file:///abs/path/to/some/notebook?:My:Page:Is:Here` links when given on the command line (see [here](https://github.com/zim-desktop-wiki/zim-desktop-wiki/blob/a637e2217b4ed1d3dbee3bcf9327d1e1f6085def/zim/notebook/__init__.py#L76) for handling) but I think it's cumbersome having to specify the full path of the notebook each time.

Rather, I'd suggest introducing a new URL scheme just for that, similar to most other URLs like so, and URL-encoded:
`zim://notebook%20name/My/Page/Is/Here#anchor2`
The implementation for this would be around 10 lines in `build_notebook`, plus some adaptations in the zim `.desktop` file to attach zim to `zim://` links. Not sure what'd be needed for Windows and macOS.

Alternatively, a tiny wrapper script could be used to handle these things too, but I think it'd be nice if zim itself were able to handle this.

What do you think? In particular it's probably relevant what @jaap-karssenberg thinks of this as he'd be the one who'd need to merge it! If there's some sort of general consensus this would be an interesting feature in zim, I can implement it and create a pull request for it :)
